src/test/java/jewellery/inventory/unit/service/security/AuthServiceTest.java

```diff
@@ -0,0 +1,112 @@
+
+import java.util.Optional;
+import java.util.UUID;
+import jewellery.inventory.dto.request.AuthenticationRequestDto;
+import jewellery.inventory.dto.response.UserAuthDetailsDto;
+import jewellery.inventory.dto.response.UserResponseDto;
+import jewellery.inventory.exception.not_found.UserNotFoundException;
+import jewellery.inventory.exception.security.InvalidCredentialsException;
+import jewellery.inventory.mapper.UserMapper;
+import jewellery.inventory.model.User;
+import jewellery.inventory.repository.UserRepository;
+import jewellery.inventory.security.JwtTokenService;
+import jewellery.inventory.service.AuthService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+
+@ExtendWith(MockitoExtension.class)
+class AuthServiceTest {
+  private static final String TEST_TOKEN = "token";
+  @InjectMocks private AuthService authService;
+  @Mock private UserRepository userRepository;
+  @Mock private AuthenticationManager authenticationManager;
+  @Mock private JwtTokenService jwtTokenService;
+  @Mock private UserMapper userMapper;
+
+  private AuthenticationRequestDto authRequest;
+  private User user;
+
+  @BeforeEach
+  public void setUp() {
+    authRequest = new AuthenticationRequestDto();
+    authRequest.setEmail(USER_EMAIL);
+    authRequest.setPassword(USER_PASSWORD);
+
+    user = new User();
+    user.setId(UUID.randomUUID());
+    user.setEmail(USER_EMAIL);
+    user.setPassword(USER_PASSWORD);
+  }
+
+  @Test
+  void authenticateWithValidCredentialsReturnsToken() {
+    when(authenticationManager.authenticate(any())).thenReturn(null);
+    when(userRepository.findByEmail(authRequest.getEmail())).thenReturn(Optional.of(user));
+    when(jwtTokenService.generateToken(user)).thenReturn(TEST_TOKEN);
+
+    UserAuthDetailsDto response = authService.login(authRequest);
+
+    assertEquals(TEST_TOKEN, response.getToken());
+  }
+
+  @Test
+  void authenticateWithInvalidCredentialsThrowsInvalidCredentialsException() {
+    user.setPassword("wrongPassword");
+    when(authenticationManager.authenticate(any())).thenThrow(new InvalidCredentialsException());
+    when(userRepository.findByEmail(anyString())).thenReturn(Optional.ofNullable(user));
+
+    assertThrows(InvalidCredentialsException.class, () -> authService.login(authRequest));
+  }
+
+  @Test
+  void authenticateWithNonExistentUserThrowsUserNotFoundException() {
+    when(userRepository.findByEmail(authRequest.getEmail())).thenReturn(Optional.empty());
+
+    assertThrows(UserNotFoundException.class, () -> authService.login(authRequest));
+  }
+
+  @Test
+  void verifyMethodsCalledWithExpectedParameters() {
```

Comment:
This method name is too generic. The setup is the same as `authenticateWithValidCredentialsReturnsToken.` Should we move the verification methods to `authenticateWithValidCredentialsReturnsToken`

Reply:
Merged with authenticateWithValidCredentialsReturnsToken